### PR TITLE
Use the language search profile if available

### PR DESCRIPTION
**Pull request by micgro42**
Bug: T312853

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -96,8 +97,13 @@ <h1>New Lexeme</h1>
 					},
 					maxLemmaLength: 100,
 				};
+				const devItemSearcher = new DevItemSearcher();
 				const services = {
-					itemSearcher: new DevItemSearcher(),
+					itemSearcher: devItemSearcher,
+					languageItemSearcher: new LanguageItemSearcher(
+						devItemSearcher,
+						['default', 'language'],
```
Comment (lucaswerkmeister) [inline index.html:105]:
```suggestion
						[ 'default', 'language' ],
```

File: src/data-access/LanguageItemSearcher.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+	public constructor( itemSearcher: MwApiItemSearcher, availableSearchProfiles: string[] ) {
+		this.itemSearcher = itemSearcher;
+		this.availableSearchProfiles = availableSearchProfiles;
+	}
+
+	public async searchItems( searchTerm: string, offset?: number ): Promise<SearchedItemOption[]> {
+		const additionalParams: Record<string, string> = {};
+		if ( this.availableSearchProfiles.includes( this.LANGUAGE_PROFILE_NAME ) ) {
+			additionalParams.profile = this.LANGUAGE_PROFILE_NAME;
```
Comment (lucaswerkmeister) [inline src/data-access/LanguageItemSearcher.ts:19]: We could also move this into the constructor, right? (And then maybe even discard `this.availableSearchProfiles`.)

File: src/plugins/ItemSearchPlugin/LanguageItemSearch.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import {
+	inject,
+	InjectionKey,
+} from 'vue';
+
+export const LanguageItemSearchKey: InjectionKey<ItemSearcher> = Symbol( 'LanguageItemSearch' );
+
+export function useLanguageItemSearch(): ItemSearcher {
+	return inject( LanguageItemSearchKey, () => {
+		throw new Error( 'No ItemSearcher provided!' );
```
Comment (lucaswerkmeister) [inline src/plugins/ItemSearchPlugin/LanguageItemSearch.ts:11]: Hm, wouldn’t it be more helpful if this message still said “No **Language**ItemSearcher provided”, even if we’re using the more general `ItemSearcher` as the return type?

File: tests/unit/data-access/LanguageItemSearcher.test.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+			expectedOptionalParams: [ 42, {} ],
+		},
+		{
+			name: 'uses language profile if available',
+			availableSearchProfiles: [ 'language', 'default' ],
+			offset: undefined,
+			expectedOptionalParams: [ undefined, { profile: 'language' } ],
+		},
+	] )( '$name', async ( { availableSearchProfiles, offset, expectedOptionalParams } ) => {
+		// const availableSearchProfiles = [ 'language', 'default' ];
```
Comment (lucaswerkmeister) [inline tests/unit/data-access/LanguageItemSearcher.test.ts:21]:
```suggestion
```

**Review by lucaswerkmeister — COMMENTED** [review]
LGTM overall

File: src/data-access/LanguageItemSearcher.ts (hunk 2 above)
Comment (micgro42) [reply src/data-access/LanguageItemSearcher.ts:19]: Not sure if I fully understand what you are suggesting. I adjusted it a bit. Let me know what you think

File: src/plugins/ItemSearchPlugin/LanguageItemSearch.ts (hunk 3 above)
Comment (micgro42) [reply src/plugins/ItemSearchPlugin/LanguageItemSearch.ts]: Done

File: src/data-access/LanguageItemSearcher.ts (hunk 2 above)
Comment (lucaswerkmeister) [reply src/data-access/LanguageItemSearcher.ts:19]: What I had in mind was that even the `additionalParams` could be a class member, but I’m also okay with just having the `useLanguageProfile` boolean there.

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**